Delete deprecated apis

krousey opened the pull request:
A work in progress for removing v1beta1 and v1beta2 APIs from the code base. #8087

Commit:
Delete deprecated API versions

pkg/service:

There were a couple of references here just as a reminder to change the
behavior of findPort. As of v1beta3, TargetPort was always defaulted, so
we could remove findDefaultPort and related tests.

pkg/apiserver:

The tests were using versioned API codecs for some of their encoding
tests. Necessary API types had to be written and registered with the
fake versioned codecs.

pkg/kubectl:

Some tests were converted to current versions where it made sense.

bgrant0607 commented on pkg/api/serialization_test.go:
```diff
@@ -103,8 +95,6 @@ func roundTripSame(t *testing.T, item runtime.Object, except ...string) {
 func roundTripAll(t *testing.T, item runtime.Object) {
 	seed := rand.Int63()
-	roundTrip(t, v1beta1.Codec, fuzzInternalObject(t, "v1beta1", item, seed))
-	roundTrip(t, v1beta2.Codec, fuzzInternalObject(t, "v1beta2", item, seed))
```
We should be roundtripping v1 here, though that could be done in another PR.

bgrant0607 commented on pkg/kubectl/cmd/util/helpers_test.go:
```diff
@@ -137,7 +100,7 @@ func TestMerge(t *testing.T) {
 				Spec: api.ServiceSpec{},
 			},
-			fragment: `{ "apiVersion": "v1beta1", "port": 0 }`,
+			fragment: `{ "apiVersion": "v1beta3", "port": 0 }`,
```
There's no port field there in v1beta3. Firstly, there's a spec. Secondly, ports are an array.

krousey replied:
Fixed.

bgrant0607 commented on pkg/service/endpoints_controller.go:
```diff
@@ -398,32 +390,14 @@ func (e *EndpointController) checkLeftoverEndpoints() {
-// is defined, use the service's port.  If the targetPort is an empty string use
-// the first defined port with the same protocol.  If no port is defined, use
-// the service's port.  If the targetPort is a non-empty string, look that
+// findPort locates the container port for the given pod and portName.  If the
```
@thockin Is this the expected behavior in v1beta3 and v1?

bgrant0607 replied:
Based on in-person discussion, this seems ok. 

TargetPort is initialized here:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/v1beta3/defaults.go#L91

thockin replied:
LGTM

thockin replied:
NB: we still have a conflict here if the same port name is used on two containers, but thta is not this PR's problem.

bgrant0607 commented:
What else needs to be done in this PR?

Please rebase at some point so we can get Travis and Shippable to run the tests. You're getting a merge conflict right now.

krousey commented:
I rebased and had to modify the code gen tools.

krousey commented:
Note, I expect on the the travis tests to fails because I haven't updated it to not use v1beta1.

bgrant0607 commented:
@krousey Please replace v1beta1 tests with v1 tests on Travis/Shippable in this PR, assuming that works. If not, just make both builds run v1beta3 tests for now.

bgrant0607 commented:
cc @smarterclayton, so he can verify we're not nuking something Openshift needs.

bgrant0607 commented on pkg/api/serialization_test.go:
```diff
@@ -137,10 +127,7 @@ func TestList(t *testing.T) {
-	"PodTemplate":     {"v1beta1", "v1beta2"},
-	"PodTemplateList": {"v1beta1", "v1beta2"},
-}
+var nonRoundTrippableTypesByVersion = map[string][]string{}
```
I suggest leaving a comment about the expected format.

bgrant0607 commented:
LGTM once tests pass.

bgrant0607 commented:
Remove WIP from the PR title once tests are working.

smarterclayton commented:
@derekwaynecarr can you double check that we're not dependent on any of these types still (pretty sure, just want to verify).

ixdy commented:
@k8s-bot test this please

(I accidentally wiped the output directory while it was building which caused it to fail.)

krousey commented:
@bgrant0607 I'll also wait until I get e2e running locally. Need to wait on a heapster release that @vishh is working on.

bgrant0607 commented:
Awesome. LGTM once e2e is working.

thockin commented:
This PR is a breath of fresh air.

bgrant0607 commented:
Yes, I am SO dying to nuke v1beta1/2.

krousey commented:
@vishh is out today, so I asked @vmarmol if he could look into getting a new heapster image released today.

krousey commented:
https://github.com/GoogleCloudPlatform/heapster/pull/301 FYI

krousey commented:
I rebased and added a couple of more commits. One removes swagger specs. I'd like someone to make sure that looks sane. The other updates our use of heapster to version 0.13.0 which was cut to add v1beta3 support.

krousey commented:
I'll squash once I get the OK.

vmarmol commented:
I would say to split some of those commits into other PRs, but outside of that the Heapster changes LGTM :)

bgrant0607 commented:
One PR/commit makes it easier to rollback.

bgrant0607 commented:
LGTM. To be merged on Monday morning.
cc @erictune, who will be on call

krousey commented:
Squashed

krousey commented:
e2e passed with new heapster image

bgrant0607 commented:
please rebase

krousey commented:
Done, and why isn't shippable running?

bgrant0607 commented:
Merge conflict would be my guess.

krousey commented:
The experimental Jenkins builder seems to have failed to fetch this PR to test. Doesn't seem to be a problem with this PR specifically.
